### DISCUSSION TITLE
Browser page refresh

### DISCUSSION THREAD

**Discussion by igiona** [unoplatform/uno#12362] [Q&A] [answered]
My WASM application uses Navigation.
When the page loads, the URL changes to the navigation destination such as `https://mysite.com/Main`
If the user then hit the refresh button, the nginx server shows a page not found error.
My location setting on the server is:

```
location / {
    try_files $uri $uri/ =404;
    autoindex on;
}
```

Is there any particular setting that can be used to avoid this, either on Uno or on the server side?
In the nginx sample here https://platform.uno/docs/articles/how-to-host-a-webassembly-app.html I don't see something relevant.
What am I missing?

Thanks!

**Comment by jeromelaban** [comment, marked as answer]
You may want to take a look at this particular document: https://platform.uno/docs/articles/external/uno.wasm.bootstrap/doc/features-deep-linking.html

**Reply by igiona** [reply]
Thanks, enabling the deep link as mentioned there worked!

**Comment by igiona** [comment]
It turned out that the solution above works well when running the app from VisualStudio, but it doesn't actually work when the app is deployed on my web server (nginx).
@jeromelaban are you aware of any special settings that the server must have in order for this to work (other then https://platform.uno/docs/articles/how-to-host-a-webassembly-app.html?q=nginx#nginx)?

**Reply by jeromelaban** [reply]
We do not have specifics for this particular web server. It's likely that you'll need to enable some sort of redirection similar to the one defined [here for azure static apps](https://platform.uno/docs/articles/guides/azure-static-webapps.html?q=static#configuring-deep-linking-with-fallback-routes). If you find a solution, let us know so we can enhance the documentation!

**Comment by KWodarczyk** [comment]
So the solution needs to be server depended there is no way to handle this in the code ? e.g. in App.xaml.cs ?
